### PR TITLE
Run a dynamic simulation using the CRV file

### PR DESCRIPTION
**Pull request by marcosmc**
**Please check if the PR fulfills these requirements** *(please use `'[x]'` to check the checkboxes, or submit the PR and then click the checkboxes)*
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** *(Bug fix, feature, docs update, ...)*
Feature

**What is the new behavior (if this is a feature change)?**
Create a CRV XML file from the configuration and update the main to run a dynamic simulation using the CRV file

### CODE REVIEW

File: src/main/java/com/powsybl/dynawo/Main.java [hunk 1]
```diff
@@ -54,10 +59,26 @@ public static void main(String[] args) {
...
+    private static String getParametersFile(String[] args) {
+        String parametersFile = args.length == 4 ? args[3] : null;
+        if (parametersFile == null) {
+            parametersFile = args.length == 3 & args[2].endsWith(".json") ? args[2] : null;
+        }
+        return parametersFile;
+    }
+
+    private static CurvesSupplier getCurvesSupplier(String[] args) {
+        if (args.length == 3 & args[2].endsWith(".crv")) {
```
Comment (agnesLeroy) [inline src/main/java/com/powsybl/dynawo/Main.java:78]: Here you need to handle the case also where there are parameters passed as argument besides the curves

File: src/main/java/com/powsybl/dynawo/Main.java [hunk 2]
```diff
@@ -54,10 +59,26 @@ public static void main(String[] args) {
...
         } catch (IOException e) {
             LOGGER.error(e.toString(), e);
             System.exit(1);
         }
     }
+
+    private static String getParametersFile(String[] args) {
+        String parametersFile = args.length == 4 ? args[3] : null;
+        if (parametersFile == null) {
+            parametersFile = args.length == 3 & args[2].endsWith(".json") ? args[2] : null;
```
Comment (agnesLeroy) [inline src/main/java/com/powsybl/dynawo/Main.java:72]: Here you're doing a bitwise "and", shouldn't it be args.length == 3 && args[2]...?
Same remark for the other condition in getCurvesSupplier.

File: src/main/java/com/powsybl/dynawo/simulator/DynawoSimulationProvider.java [hunk 3]
```diff
@@ -87,13 +89,12 @@ private DynawoSimulationParameters getDynawoSimulationParameters(DynamicSimulati
         return dynawoParameters;
     }
 
-    private CompletableFuture<DynamicSimulationResult> run(Network network, String workingStateId, ComputationManager computationManager,
-        DynamicSimulationParameters parameters, DynawoSimulationParameters dynawoParameters) {
+    private CompletableFuture<DynamicSimulationResult> run(Network network, CurvesSupplier curvesSupplier, String workingVariantId, ComputationManager computationManager, DynamicSimulationParameters parameters, DynawoSimulationParameters dynawoParameters) {
```
Comment (agnesLeroy) [inline src/main/java/com/powsybl/dynawo/simulator/DynawoSimulationProvider.java:92]: It seemed easier to read with the arguments on two lines, could you place part of them on a second line?

File: src/test/java/com/powsybl/dynawo/xml/CurvesXmlTest.java [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+        DynawoSimulationParameters dynawoParameters = DynawoSimulationParameters.load();
+
+        Network network = Network.create("test", "test");
+
+        List<Curve> curves = Arrays.asList(new DynawoCurve("NETWORK", "busId_Upu_value"),
+                                            new DynawoCurve("genId", "generator_omegaPu"),
+                                            new DynawoCurve("genId", "generator_PGen"),
+                                            new DynawoCurve("genId", "generator_UStatorPu"),
+                                            new DynawoCurve("genId", "voltageRegulator_UcEfdP"),
+                                            new DynawoCurve("genId", "voltageRegulator_EfdPu"));
```
Comment (agnesLeroy) [inline src/test/java/com/powsybl/dynawo/xml/CurvesXmlTest.java:42]: In this test it should be made clear whether the genId is a static ID or a dynamic model ID: genId could be both.

File: src/test/java/com/powsybl/dynawo/xml/JobsXmlTest.java [hunk 5]
```diff
@@ -30,13 +34,13 @@ public void writeJob() throws SAXException, IOException, XMLStreamException {
...
-            public boolean withCurves() {
-                return true;
-            }
-        };
+        List<Curve> curves = Arrays.asList(new DynawoCurve("NETWORK", "busId_Upu_value"),
+            new DynawoCurve("genId", "generator_omegaPu"),
+            new DynawoCurve("genId", "generator_PGen"),
+            new DynawoCurve("genId", "generator_UStatorPu"),
+            new DynawoCurve("genId", "voltageRegulator_UcEfdP"),
+            new DynawoCurve("genId", "voltageRegulator_EfdPu"));
```
Comment (agnesLeroy) [inline src/test/java/com/powsybl/dynawo/xml/JobsXmlTest.java:42]: Same remark about the "genId" here.

File: src/test/resources/curvesInput.xml [hunk 6]
```diff
@@ -0,0 +1,9 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<curvesInput xmlns="http://www.rte-france.com/dynawo">
+  <curve model="NETWORK" variable="busId_Upu_value"/>
+  <curve model="genId" variable="generator_omegaPu"/>
+  <curve model="genId" variable="generator_PGen"/>
+  <curve model="genId" variable="generator_UStatorPu"/>
+  <curve model="genId" variable="voltageRegulator_UcEfdP"/>
+  <curve model="genId" variable="voltageRegulator_EfdPu"/>
+</curvesInput>
```
Comment (agnesLeroy) [inline src/test/resources/curvesInput.xml:9]: Same remark

File: src/main/java/com/powsybl/dynawo/xml/CurvesXml.java [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+        Objects.requireNonNull(workingDir);
+        Objects.requireNonNull(context);
+        Path file = workingDir.resolve(CRV_FILENAME);
+        try (Writer writer = Files.newBufferedWriter(file, StandardCharsets.UTF_8)) {
+            XMLStreamWriter xmlWriter = XmlStreamWriterFactory.newInstance(writer);
+            try {
+                xmlWriter.writeStartDocument(StandardCharsets.UTF_8.toString(), "1.0");
+                xmlWriter.setPrefix(EMPTY_PREFIX, DYN_URI);
+                xmlWriter.writeStartElement(DYN_URI, "curvesInput");
+                xmlWriter.writeNamespace(EMPTY_PREFIX, DYN_URI);
```
Comment (mathbagu) [inline src/main/java/com/powsybl/dynawo/xml/CurvesXml.java:45]: I think we should write the curves with the `dyn` prefix, like the other files.

File: src/test/resources/curvesInput.xml (hunk 6 above)
Comment (mathbagu) [reply src/test/resources/curvesInput.xml:9]: Could you add an empty line at the end of this file?

File: src/test/resources/curvesInput.xsd [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+      <xs:field xpath="@model"/>
+      <xs:field xpath="@variable"/>
+    </xs:unique>
+  </xs:element>
+
+  <xs:complexType name="CurveInput">
+    <xs:attribute name="model" use="required" type="xs:string"/>
+    <xs:attribute name="variable" use="required" type="xs:string"/>
+  </xs:complexType>
+</xs:schema>
```
Comment (mathbagu) [inline src/test/resources/curvesInput.xsd:35]: Could you add an empty line at the end of this file?

File: src/test/java/com/powsybl/dynawo/xml/CurvesXmlTest.java (hunk 4 above)
Comment (mathbagu) [reply src/test/java/com/powsybl/dynawo/xml/CurvesXmlTest.java:42]: I think your remark could be applied to the busId. @marcosmc, maybe you should use real name even it's a test case. Try to find the modelID of the network that has these curves available. Maybe use a IIDM file from dynawo test case to find real static IDs

File: src/main/java/com/powsybl/dynawo/Main.java (hunk 1 above)
Comment (marcosmc) [reply src/main/java/com/powsybl/dynawo/Main.java]: Done

File: src/main/java/com/powsybl/dynawo/Main.java (hunk 2 above)
Comment (marcosmc) [reply src/main/java/com/powsybl/dynawo/Main.java]: Done

File: src/main/java/com/powsybl/dynawo/simulator/DynawoSimulationProvider.java (hunk 3 above)
Comment (marcosmc) [reply src/main/java/com/powsybl/dynawo/simulator/DynawoSimulationProvider.java]: Done

File: src/test/java/com/powsybl/dynawo/xml/CurvesXmlTest.java (hunk 4 above)
Comment (marcosmc) [reply src/test/java/com/powsybl/dynawo/xml/CurvesXmlTest.java]: Changed to use EurostagTutorialExample1Factory and its static IDs

File: src/test/java/com/powsybl/dynawo/xml/JobsXmlTest.java (hunk 5 above)
Comment (marcosmc) [reply src/test/java/com/powsybl/dynawo/xml/JobsXmlTest.java]: Changed to use EurostagTutorialExample1Factory and its static IDs

File: src/test/resources/curvesInput.xml (hunk 6 above)
Comment (marcosmc) [reply src/test/resources/curvesInput.xml]: Changed to use EurostagTutorialExample1Factory and its static IDs

File: src/test/resources/curvesInput.xsd (hunk 8 above)
Comment (marcosmc) [reply src/test/resources/curvesInput.xsd]: Done

File: src/test/resources/curvesInput.xml (hunk 6 above)
Comment (marcosmc) [reply src/test/resources/curvesInput.xml]: added an empty line at the end of the file

File: src/test/java/com/powsybl/dynawo/xml/DynawoTestUtil.java [hunk 9]
```diff
@@ -17,15 +19,39 @@
...
+        network = EurostagTutorialExample1Factory.create();
+
+        curves = new ArrayList<>();
+        network.getBusBreakerView().getBusStream().forEach(b -> curves.add(new DynawoCurve("NETWORK", b.getId() + "_Upu_value")));
+        network.getGeneratorStream().forEach(g -> {
+            curves.add(new DynawoCurve(g.getId(), "generator_omegaPu"));
+            curves.add(new DynawoCurve(g.getId(), "generator_PGen"));
+            curves.add(new DynawoCurve(g.getId(), "generator_UStatorPu"));
+            curves.add(new DynawoCurve(g.getId(), "voltageRegulator_UcEfdP"));
+            curves.add(new DynawoCurve(g.getId(), "voltageRegulator_EfdPu"));
```
Comment (agnesLeroy) [inline src/test/java/com/powsybl/dynawo/xml/DynawoTestUtil.java:50]: Hum, I don't think that's quite it: the ID should be one of a dynamic model, not a static id here (the same remark goes for the other places where the modelId is involved).

Comment (agnesLeroy) [reply src/test/java/com/powsybl/dynawo/xml/DynawoTestUtil.java:50]: For the bus with NETWORK dynamic model I think what you did is correct, but for the generator it should be directly the dynamic model ID associated to the generator in the first argument of DynawoCurve.

Comment (agnesLeroy) [reply src/test/java/com/powsybl/dynawo/xml/DynawoTestUtil.java:50]: @rosiereflo could you indicate us what to use here please?

File: src/main/java/com/powsybl/dynawo/xml/CurvesXml.java (hunk 7 above)
Comment (marcosmc) [reply src/main/java/com/powsybl/dynawo/xml/CurvesXml.java]: Done

**Pull request merged**